libs/libc/misc/lib_getnprocs.c

```diff
@@ -0,0 +1,79 @@
+/****************************************************************************
+ * libs/libc/misc/lib_getnprocs.c
+ *
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.  The
+ * ASF licenses this file to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance with the
+ * License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.  See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ *
+ ****************************************************************************/
+
+/****************************************************************************
+ * Included Files
+ ****************************************************************************/
+
+#include <nuttx/config.h>
+#include <sys/sysinfo.h>
+#include <sys/types.h>
```

Comment:
Why do we need to include `sys/types.h`?

Reply:
If we don't configure CONFIG_SMP_NCPUS in config, then it won't find this definition. Instead, CONFIG_SMP_NCPUS is the default configured in <sys/types.h>.

```
misc/lib_getnprocs.c: In function ‘get_nprocs_conf’:
misc/lib_getnprocs.c:54:10: error: ‘CONFIG_SMP_NCPUS’ undeclared (first use in this function); did you mean ‘CONFIG_NET_TCP’?
   54 |   return CONFIG_SMP_NCPUS;
      |          ^~~~~~~~~~~~~~~~
      |          CONFIG_NET_TCP
misc/lib_getnprocs.c:54:10: note: each undeclared identifier is reported only once for each function it appears in
misc/lib_getnprocs.c: In function ‘get_nprocs’:
misc/lib_getnprocs.c:78:10: error: ‘CONFIG_SMP_NCPUS’ undeclared (first use in this function); did you mean ‘CONFIG_NET_TCP’?
   78 |   return CONFIG_SMP_NCPUS;
```